crates/krun/src/bin/krun.rs

```diff
@@ -58,10 +60,17 @@ fn main() -> Result<()> {
     };
 
     {
-        // Configure the number of vCPUs (4) and the amount of RAM (4096 MiB).
+        let ram_mib = if let Some(ram_mib) = options.mem {
+            ram_mib
+        } else {
+            let sysinfo = sysinfo().context("Failed to get system information")?;
+            let ram_total = sysinfo.ram_total() / 1024 / 1024;
+            cmp::min(MiB::from((ram_total as f64 * 0.8) as u32), MiB::from(16384))
+        };
+        // Configure the number of vCPUs (4) and the amount of RAM (max 16384 MiB).
```

Comment:
(From https://github.com/teohhanhui/krun/commit/ce29e32992389569b70076ccc791577b8d0cfe28#r142414630)

@asahilina
> Is there a reason for this limit? We have machines with 192GB of RAM, surely it makes sense for the VM to support more than 16G...

I have no idea why there's this limit in krunvm. Was going to ask @slp since he's the original author of that code according to git blame. Will remove once we confirm that it's not needed.